Bug 1412824 Added context to egress router section

bfallonf opened the pull request:
As per: https://bugzilla.redhat.com/show_bug.cgi?id=1412824

I'll ask for an ack in the BZ.

cc: @sferich888

sferich888 commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,17 @@ endif::openshift-enterprise,openshift-origin[]
+As an {product-title} cluster administrator, you can control egress traffic in two different ways:
+
+* Using an egress xref:admin-guide-deploying-an-egress-router-service[router], so that an external source treats traffic as though it were coming from a known source.
+* Using an egress xref:admin-guide-limit-pod-access-egress[firewall], where traffic is accepted, enforced, and defined.
```
I think this is trying to say, what https://docs.openshift.com/container-platform/3.3/admin_guide/managing_pods.html#admin-guide-limit-pod-access-egress says, but it does not quite align. 

In short `accepted, enforced, and defined` (are not how I would phrase this). 

In stead I would say: 

`Using an egress xref:admin-guide-limit-pod-access-egress[firewall] allows a cluster administrator to enforce acceptable (outbound) traffic policies. These policies can be defined in such a way that, acceptable endpoints or ip ranges (subnets) are the only acceptable targets for the dynamic endpoints (pods with in OpenShift) to talk to.`

sferich888 commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,17 @@ endif::openshift-enterprise,openshift-origin[]
+As an {product-title} cluster administrator, you can control egress traffic in two different ways:
+
+* Using an egress xref:admin-guide-deploying-an-egress-router-service[router], so that an external source treats traffic as though it were coming from a known source.
+* Using an egress xref:admin-guide-limit-pod-access-egress[firewall], where a cluster administrator can enforce acceptable (outbound) traffic policies, so that specific endpoints or IP ranges (subnets) are the only acceptable targets for the dynamic endpoints (pods within {product-title}) to talk to.
```
`Using an egress xref:admin-guide-limit-pod-access-egress[firewall], where a cluster administrator` does not read right. Should this say `Using an egress xref:admin-guide-limit-pod-access-egress[firewall], allows a cluster administrator`?

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,126 @@ endif::openshift-enterprise,openshift-origin[]
+
+xref:admin-guide-limit-pod-access-egress-router[Router]::
+Using an egress router allows you to create identifiable services to send
+traffic to a specific source, ensuring an external source treats traffic as
```
"destination" not "source" (both times)

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,126 @@ endif::openshift-enterprise,openshift-origin[]
+Violating any of these restrictions results in broken egress policy for the
+project, and may cause all external network traffic to be dropped.
+
+Use the `oc` command or the REST API to configure pod access limits. You can use
```
Probably it's best to get rid of the "pod access limits" language altogether? "to configure egress policy" (and then again below)

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,126 @@ endif::openshift-enterprise,openshift-origin[]
+# oc create -f <policy>.json
+----
+
+. Configure the JSON file with policy details. For example:
```
This is backwards... you have to create the JSON file first, then call "oc create -f <policy>.json" (which reads *from* the file, to create the object in OpenShift)

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -219,61 +331,45 @@ endif::openshift-origin[]
 [NOTE]
 ====
 Preserve the the quotation marks around `"true"`. Omitting them will result in
 errors.
```
move this into note <1> now?

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -219,61 +331,45 @@ endif::openshift-origin[]
+<3> Same value as the default gateway used by the node itself.
+<4>  Connections to the pod are redirected to 203.0.113.25, with a source IP
+address of 192.168.12.99
+<5> The pod will only be deployed to nodes with the label site `springfield-1`.
```
"site" is part of the label. It should say

    ...with the label `site=springfield-1`.

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,126 @@ endif::openshift-enterprise,openshift-origin[]
+
+
+[[admin-guide-limit-pod-access-egress]]
+=== Limiting Pod Access with Egress Firewall
```
These titles (this one and "Limiting Pod Access with an Egress Router") should probably be changed too, to make it clearer that these are two fairly different things. Maybe "Limiting Access to External Resources with an Egress Firewall" and "Allowing External Resources to Recognize Pod Traffic with an Egress Router" or something

bfallonf replied:
I like it. It helps with Eric's initial BZ request about making the two procedures more different.

bfallonf commented:
Thanks, @danwinship . I've made some updates as per your comments, added the answers you gave in the BZ, and added the section on egress router failover. Can I ask for a final ack or if there are any more issues? Thanks in advance.

cc: @sferich888 as the original BZ reporter, let me know if there's anything else you'd like to add.

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,126 @@ endif::openshift-enterprise,openshift-origin[]
+xref:admin-guide-limit-pod-access-egress-router[Router]::
+Using an egress router allows you to create identifiable services to send
+traffic to a specific destination, ensuring an external source treats traffic as
+though it were coming from a known destination. This helps with security,
```
...traffic to a specific destination, ensuring an external **destination** treats traffic as though it were coming from a known **source**.

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,126 @@ endif::openshift-enterprise,openshift-origin[]
+
+. Navigate to the project you want to affect.
+
+. Configure the JSON file with policy details. For example:
```
"What JSON file?"

Should say something like: "Create a JSON file with policy details. For example:"

and then the next bullet point (which currently says "Create a JSON file...") should be something like "Apply that policy to the project" or "Create an EgressNetworkPolicy object from the JSON file".

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -219,32 +331,63 @@ endif::openshift-origin[]
+<4> The external server that to direct traffic to. Using this example, connections
+to the pod are redirected to 203.0.113.25, with a source IP address of
+192.168.12.99.
+<5> The pod will only be deployed to nodes with the label `springfield-1`.
```
\`site=springfield-1\`, not just \`springfield-1\`. This line should be exactly the following text:

    <5> The pod will only be deployed to nodes with the label `site=springfield-1`.

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -219,32 +331,63 @@ endif::openshift-origin[]
+. Create the pod using the definition:
++
+----
+$ oc create -f <pod_name>
```
You have to pass the name of the JSON file created in the previous step, not the pod name. (But in the "oc get" example below, using pod_name is correct.)

It looks like just saying "<pod_name>.json" might be consistent with other examples

danwinship commented on admin_guide/managing_pods.adoc:
```diff
@@ -268,129 +411,62 @@ If only some of the nodes in your cluster are capable of claiming the specified
 +
 ----
-# oc create -f <policy>.json
+$ oc create -f <replication_controller>
```
add ".json"

bfallonf commented:
Thanks @danwinship Let me know if there's anything else.

In the meantime, @ahardin-rh @bmcelvee @adellape peer review please?

ahardin-rh commented on admin_guide/managing_pods.adoc:
```diff
@@ -146,16 +146,138 @@ endif::openshift-enterprise,openshift-origin[]
+You can configure projects to have different egress policies. For example,
+allowing `<project A>` access to a specified IP range, but denying the same
+access to `<project B>`. Or restrict application developers from updating from
+(python) pip mirrors, and forcing updates to only come from desired sources.
```
s/python/Python

ahardin-rh commented on admin_guide/managing_pods.adoc:
```diff
@@ -219,32 +343,63 @@ endif::openshift-origin[]
+----
+<1> The `pod.network.openshift.io/assign-macvlan annotation` creates a Macvlan
+network interface on the primary network interface, and then moves it into the
+pod's network name space before starting the *egress-router* container. Preserve the the quotation marks around `"true"`. Omitting them will result in
```
Maybe wrap all of <1>

ahardin-rh commented:
@bfallonf just minor nits from me :ok_hand:

bfallonf commented:
Thanks @ahardin-rh . I'll merge.

bfallonf commented:
[rev_history]
|xref:../admin_guide/managing_pods.adoc#admin-guide-manage-pods[Managing Pods]
|Added information and arranged the  xref:../admin_guide/managing_pods.adoc#admin-guide-controlling-egress-traffic[Controlling Egress Traffic] section.
%
